GPflow/gpr.py

```diff
@@ -46,15 +46,19 @@ def build_predict(self, Xnew):
         where F* are points on the GP at Xnew, Y are noisy observations at X.
 
         """
-        Kd = self.kern.Kdiag(Xnew)
         Kx = self.kern.K(self.X, Xnew)
         K = self.kern.K(self.X) + eye(self.num_data) * self.likelihood.variance
         L = tf.cholesky(K)
         A = tf.matrix_triangular_solve(L, Kx, lower=True)
         V = tf.matrix_triangular_solve(L, self.Y - self.mean_function(self.X), lower=True)
         fmean = tf.matmul(tf.transpose(A), V) + self.mean_function(Xnew)
-        fvar = Kd - tf.reduce_sum(tf.square(A), reduction_indices=0)
-        return fmean, tf.tile(tf.reshape(fvar, (-1,1)), [1, self.Y.shape[1]])
+        if full_cov:
+            fvar = self.kern.K(Xnew) - tf.matmul(tf.transpose(A), A)
```

Comment:
So, the following methods use conditionals.py for predictions:
 - GPMC (Full covariance MCMC)
 - SGPMC (Sparse MCMC)
 - SVGP (Sparse variational GP (non-conjugate))

These methods do not:
 - GPR (Regression with Gaussian noise). 
 - SGPR (Sparse Regression)
 - VGP (Variational GP)

The reason is that for these methods, computing the posterior and then computing the corresponding conditional would introduce redundant computation.